frontend/src/components/TextField/TextField.tsx

```diff
@@ -1,48 +1,67 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable react/jsx-no-bind */
 import React from 'react';
 import { TextField as MuiTextField } from '@mui/material';
-import { Field } from 'formik';
+import { Field, ErrorMessage } from 'formik';
+
+import WarningHelper from './styled';
 
 interface TextFieldProps {
   id: string;
-  label: string;
+  handleChange: (e: string) => void;
+  isFieldTouched: boolean;
+  setFieldTouched: (name: string, value?: boolean, isValidate?: boolean) => void;
+  label?: string;
   children?: React.ReactNode;
   defaultValue?: string;
   helperText?: string;
   touched?: boolean;
+  error?: {
+    login?: string;
+    password?: string;
+  };
   onChange?: (e: string) => void;
-  error?: string;
+  handleBlur?: (e: React.FocusEvent) => void;
   value?: unknown;
   type?: string;
+  placeholder?: string;
 }
 
 const TextField: React.FC<TextFieldProps> = ({
   id,
+  handleChange,
+  isFieldTouched,
+  setFieldTouched,
   label,
   children,
   defaultValue,
   helperText,
-  onChange,
-  error,
   touched,
+  error,
+  onChange,
+  handleBlur,
   ...rest
 }) => (
   <>
     <Field
       component={MuiTextField}
-      onChange={onChange}
+      onChange={(e: string) => {
+        handleChange(e);
+        setFieldTouched(id, isFieldTouched);
```

Comment:
Move it please inside separate common function and call it here:
```suggestion
        onChange={onChange}
```

Reply:
Created new function for this and filled it in Container